Automattic/wp-calypso
# Subscribers page: add search component & context

Closes #78359

## Proposed Changes

This PR adds the search bar. The search bar is shown when the total number of subscribers > 0. 

Besides that, it also adds a context, responsible for storing data shared between the components.

## Testing Instructions
1. Apply this PR
2. Verify that the search bar shows on `http://calypso.localhost:3000/subscribers/<siteid>`, the design is [correct](h8tMpJlCqNFemEerU9owHI-fi-1984_79028) & there are no regressions 


## Pre-merge Checklist
